dbsim/db_high_priority_service.cpp

```diff
@@ -84,24 +86,39 @@ int db::high_priority_service::rollback(const wsrep::ws_handle& ws_handle,
     return ret;
 }
 
+void db::high_priority_service::adopt_apply_error(
+    const wsrep::mutable_buffer& err)
+{
+    client_.client_state_.adopt_apply_error(err);
+}
+
 void db::high_priority_service::after_apply()
 {
     client_.client_state_.after_applying();
 }
 
 int db::high_priority_service::log_dummy_write_set(
     const wsrep::ws_handle& ws_handle,
-    const wsrep::ws_meta& ws_meta)
+    const wsrep::ws_meta& ws_meta,
+    const wsrep::mutable_buffer& err)
 {
     int ret(client_.client_state_.start_transaction(ws_handle, ws_meta));
     assert(ret == 0);
-    client_.client_state_.prepare_for_ordering(ws_handle, ws_meta, true);
-    ret = client_.client_state_.before_commit();
-    assert(ret == 0);
-    ret = client_.client_state_.ordered_commit();
-    assert(ret == ret);
-    ret = client_.client_state_.after_commit();
-    assert(ret == 0);
+    if (ws_meta.ordered())
+    {
+        client_.client_state_.prepare_for_ordering(ws_handle, ws_meta, true);
+        ret = client_.client_state_.before_commit();
+        assert(ret == 0);
+        ret = client_.client_state_.ordered_commit();
+        assert(ret == 0);
+        ret = client_.client_state_.after_commit();
+        assert(ret == 0);
+        ret = client_.client_state_.provider().commit_order_enter(
```

Comment:
Explicit `commit_order_enter()`/`commit_order_leave()` is not necessary. If `prepare_for_ordering()` is called with ordered `ws_meta`, the `before_commit()`/`ordered_commit()`/`after_commit()` sequence acquires and releases commit monitor internally.

Reply:
fixed